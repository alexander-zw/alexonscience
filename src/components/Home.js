/**
 * The home component is the default component displayed at the index URL.
 * It contains a basic welcome.
 */
import React from "react";
import { Helmet } from "react-helmet";
import { Link } from "react-router-dom";
import "../styles/index.css";
import ExpansionText from "./subcomponents/ExpansionText";

function Home() {
    const expansionText =
        "As you might be able to tell, I love drawing, even if I'm not that good at " +
        "it. Here you can see a Turing machine, AlphaGo, Albert Einstein, some DNA, " +
        "and Rosalind Franklin. Franklin is particularly underappreciated - she was " +
        "not recognized for a Nobel prize only because she had passed away by the time " +
        "it was awarded.";

    const expansionTextComponent = <p>{expansionText}</p>;

    return (
        <div>
            <Helmet>
                <title>ALEX on Science</title>
                <meta name="Description" content="The official ALEX on Science website" />
                <meta name="KeyWords" content="alex, alexander, wu, science, youtube" />
            </Helmet>

            <div className="text-div bottom-margin">
                <p>
                    Hello there! My name is Alexander Wu. I am a graduate student who enjoys
                    maintaining a YouTube channel and working on personal coding projects. Feel free
                    to explore around!
                </p>

                <p>
                    Check out some of my awesome <Link to="/projects">projects</Link>! Or, if you
                    {"'"}d like to learn about me (I{"'"}m flattered), take a look at my{" "}
                    <Link to="/resume">resume</Link>.
                </p>

                <ExpansionText expansionComponent={expansionTextComponent}>
                    What are those drawings in your banner?
                </ExpansionText>
            </div>
        </div>
    );
}

export default Home;
