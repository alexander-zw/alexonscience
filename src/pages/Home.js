import React from 'react';
import '../styles/index.css';

function Home() {
    return (
        <div>
            <div className="text-div">
                <p>Psst... check out my <a href="https://tinyurl.com/alexonscience">Youtube channel</a>!</p>
            </div>
        </div>
    );
}

export default Home;
