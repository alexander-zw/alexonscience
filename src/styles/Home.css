/* Popup container. */
.popup {
    position: relative;
    display: inline-block;
    cursor: pointer;
    z-index: 9999;
}

/* Text you're supposed to click. */
.popup-outer {
    font-size: 0.9em;
    color: rgb(99, 99, 99);
}

/* The actual popup (appears on top). */
.popup .popuptext {
    visibility: hidden;
    width: 18em;
    background-color: rgb(100, 100, 100);
    color: #fff;
    text-align: center;
    font-size: 0.85em;
    border-radius: 0.7em;
    padding: 0.5em 0.5em 0.5em 0.5em;
    position: absolute;
    z-index: 1;
    bottom: 125%;
    left: 50%;
    margin-left: -9.5em;
}

/* Popup arrow. */
.popup .popuptext::after {
    content: "";
    position: absolute;
    top: 100%;
    left: 50%;
    margin-left: -5px;
    border-width: 5px;
    border-style: solid;
    border-color: #555 transparent transparent transparent;
}

/* Hide and show the popup. */
.popup .show {
    visibility: visible;
    -webkit-animation: fadeIn 1s;
    animation: fadeIn 1s;
}

@-webkit-keyframes fadeIn {
    from {opacity: 0;}
    to {opacity: 1;}
}

@keyframes fadeIn {
    from {opacity: 0;}
    to {opacity: 1;}
}
