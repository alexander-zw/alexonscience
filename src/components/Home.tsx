import "../styles/index.css";
import "../styles/Home.css";

import React from "react";
import { Link } from "react-router-dom";

import ExpansionText from "./subcomponents/ExpansionText";
import MetaTags from "./subcomponents/MetaTags";
import RecentVideos from "./subcomponents/RecentVideos";

/**
 * The home component is the default component displayed at the index URL.
 * It contains a basic welcome.
 */
function Home() {
    const expansionText =
        "As you might be able to tell, I love drawing, even if I'm not that good at " +
        "it. Here you can see a Turing machine, AlphaGo, Albert Einstein, some DNA, " +
        "and Rosalind Franklin. Franklin is particularly underappreciated - she was " +
        "not recognized for a Nobel prize only because she had passed away by the time " +
        "it was awarded.";

    return (
        <div>
            <MetaTags path="/" />

            <div className="text-div bottom-margin">
                <p>
                    Hello there! My name is Alexander Wu. When I{"'"}m not working, I enjoy
                    maintaining a YouTube channel and working on personal coding projects. Feel free
                    to explore around!
                </p>

                <p>
                    Check out some of my awesome <Link to="/projects">projects</Link>! Or, if you
                    {"'"}d like to learn about me (I{"'"}m flattered), take a look at my{" "}
                    <Link to="/resume">resume</Link>.
                </p>

                <RecentVideos />

                <ExpansionText className="expansion" expansionComponent={expansionText}>
                    What are those drawings in your banner?
                </ExpansionText>
            </div>
        </div>
    );
}

export default Home;
