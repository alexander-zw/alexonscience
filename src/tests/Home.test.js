/**
 * Test for the Home.js component.
 */
import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Home from "../components/Home";

test("renders my name", () => {
    render(<Home />);
    const linkElement = screen.getByText(/Alexander Wu/i);
    expect(linkElement).toBeInTheDocument();
});

it("renders correctly", () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
});
