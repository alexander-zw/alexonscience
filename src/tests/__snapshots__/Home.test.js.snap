// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="text-div bottom-margin"
  >
    <p>
      Hello there! My name is Alexander Wu. I am a graduate student who enjoys maintaining a YouTube channel and working on personal coding projects. Feel free to explore around!
    </p>
    <div
      className="popup"
      onClick={[Function]}
    >
      <span
        className="popup-outer"
      >
        What are those drawings in your banner?
      </span>
      <span
        className="popuptext"
      >
        As you might be able to tell, I love drawing, even if I'm not that good at it. Here you can see a Turing machine, AlphaGo, Albert Einstein, some DNA, and Rosalind Franklin. Franklin is particularly underappreciated - she was not recognized for a Nobel prize only because she had passed away by the time it was awarded.
      </span>
    </div>
  </div>
</div>
`;
